test/destest.c

```diff
@@ -718,6 +718,31 @@ int main(int argc, char *argv[])
         printf("fast crypt error, %s should be yA1Rp/1hZXIJk\n", str);
         err = 1;
     }
+    str = crypt("testing", "y€");
+    if (str != NULL) {
+        printf("salt error only usascii are accepted, returned value should be 0\n");
+        err = 1;
+    }
+    str = crypt("testing", "‰A");
+    if (str != NULL) {
+        printf("salt error only usascii are accepted, returned value should be 0\n");
+        err = 1;
+    }
+    str = crypt("testing", "\0A");
```

Comment:
The string is **'\', '0', 'A', '\0'**, so first two characters do not match nul ('\0').